tests/lib/__init__.py

```diff
@@ -353,6 +353,38 @@ def assert_installed(self, pkg_name, editable=True, with_files=[],
                     .format(**locals())
                 )
 
+    def did_create(self, path, message=None):
+        if message is None:
+            message = str(self)
+        else:
+            message = "\n".join((message, str(self)))
```

Comment:
I see that this `if-else` is being repeated at 4 places. Maybe it's a good idea to move this in it's own function, and use a single line if-else. (Or you can just move it in your own function, and not use the single line if-else for readability)

```
def create_msg(message=None):

    return "\n".join((message, str(self))) if message else str(self)
```

Reply:
Sorry for bikeshedding, but I think it's more readable to have the messaged string as `{}\n{},format(message, self)`.  A bit more importantly, the repetition signals that the assertions are similar enough that parameterizing them might be a good idea.

Reply:
> Sorry for bikeshedding, but I think it's more readable to have the messaged string as {}\n{},format(message, self)

Agreed that `return '{}\n{}'.format(message, self) if message else self` is a better idea (We don't need to explicitly cast `self` to `str

> A bit more importantly, the repetition signals that the assertions are similar enough that parameterizing them might be a good idea.

I think it was suggested at https://github.com/pypa/pip/issues/6050#issuecomment-597743589 to have different functions for asserting `in` and `not in` of `self.files_created` and `self.files_updated`